### PR TITLE
[vcpkg baseline] Clean up baseline, use Keyword Supports Part 3

### PR DESCRIPTION
**Pull request by JackBoosY**
Separate PR #13384.

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -109,45 +109,11 @@ bitserializer-rapidyaml:arm64-windows=skip
 bitserializer-rapidyaml:x64-osx=skip
 bitserializer:arm-uwp=skip
 bitserializer:arm64-windows=skip
```
Comment (JackBoosY) [inline scripts/ci.baseline.txt:111]:
```suggestion
bitserializer:arm64-windows=skip
bitserializer:x64-osx=skip
```

**Comment by strega-nil** [comment]
Thanks @JackBoosY :)

File: ports/bond/CONTROL [hunk 2]
```diff
@@ -3,6 +3,7 @@ Version: 9.0.3
 Description: Bond is a cross-platform framework for working with schematized data. It supports cross-language de/serialization and powerful generic mechanisms for efficiently manipulating data. Bond is broadly used at Microsoft in high scale services.
 Homepage: https://github.com/Microsoft/bond
 Build-Depends: rapidjson, boost-config, boost-utility, boost-assign, boost-locale
+Supports: !(uwp|osx)
```
Comment (strega-nil) [inline ports/bond/CONTROL:6]: Why is this supports !osx? It seems that bond is fine on macOS, you just need the right system deps.

File: ports/caf/CONTROL [hunk 3]
```diff
@@ -4,3 +4,4 @@ Port-Version: 1
 Homepage: https://github.com/actor-framework/actor-framework
 Build-Depends: openssl
 Description: an open source C++11 actor model implementation featuring lightweight & fast actor implementations, pattern matching for messages, network transparent messaging, and more.
+Supports: !(arm|uwp)
```
Comment (strega-nil) [inline ports/caf/CONTROL:7]: It looks like they at least attempt to support arm, so I don't think this is a good change.

File: ports/caffe2/CONTROL [hunk 4]
```diff
@@ -3,4 +3,4 @@ Version: 0.8.1-3
 Build-Depends: lmdb, gflags, glog, eigen3, protobuf
 Homepage: https://github.com/caffe2/caffe2
 Description: Caffe2 is a lightweight, modular, and scalable deep learning framework.
-Supports: !x86
\ No newline at end of file
+Supports: !(x86|arm)
```
Comment (strega-nil) [inline ports/caffe2/CONTROL:6]: Could you add a `vcpkg_fail_port_install` or w/e it's called to the portfile as well?

File: ports/cairomm/CONTROL [hunk 5]
```diff
@@ -4,3 +4,4 @@ Port-Version: 5
 Homepage: https://www.cairographics.org
 Description: A C++ wrapper for the cairo graphics library
 Build-Depends: cairo, libsigcpp
+Supports: !linux
```
Comment (strega-nil) [inline ports/cairomm/CONTROL:7]: Why doesn't cairomm support linux? that seems really weird.

File: ports/capnproto/CONTROL [hunk 6]
```diff
@@ -3,4 +3,4 @@ Version: 0.8.0
 Description: Data interchange format and capability-based RPC system
 Homepage: https://capnproto.org/
 Build-Depends: zlib
-Supports: !uwp
\ No newline at end of file
+Supports: !(arm|uwp)
```
Comment (strega-nil) [inline ports/capnproto/CONTROL:6]: Does capnproto really not support ARM, even on non-Windows? that seems weird to me.

File: ports/cartographer/CONTROL [hunk 7]
```diff
@@ -3,3 +3,4 @@ Version: 1.0.0-3
 Build-Depends: ceres[suitesparse], gflags, glog, lua, cairo, boost-iostreams, gtest, protobuf
 Homepage: https://github.com/googlecartographer/cartographer
 Description: Google 2D & 3D SLAM package
+Supports: !osx
```
Comment (strega-nil) [inline ports/cartographer/CONTROL:6]: Why doesn't cartographer support osx?

**Review by strega-nil — COMMENTED** [review]
I just have some questions, and at least a few ones that I don't agree with.

File: ports/bond/CONTROL (hunk 2 above)
Comment (JackBoosY) [reply ports/bond/CONTROL:6]: @strega-nil Can you please take a look?
```
843 In file included from /Users/vcpkg/Documents/vcpkg/vcpkg/installed/x64-osx/include/boost/locale.hpp:22:
844 /Users/vcpkg/Documents/vcpkg/vcpkg/installed/x64-osx/include/boost/locale/localization_backend.hpp:109:18: error: 'auto_ptr<boost::locale::localization_backend>' is deprecated [-Werror,-Wdeprecated-declarations]
845             std::auto_ptr<localization_backend> get() const;
846                  ^
847 /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/memory:2135:28: note: 'auto_ptr<boost::locale::localization_backend>' has been explicitly marked deprecated here
848 class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX11 auto_ptr
849                            ^
850 /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__config:1030:39: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX11'
851 #  define _LIBCPP_DEPRECATED_IN_CXX11 _LIBCPP_DEPRECATED
852                                       ^
853 /Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__config:1019:48: note: expanded from macro '_LIBCPP_DEPRECATED'
854 #    define _LIBCPP_DEPRECATED __attribute__ ((deprecated))
855                                                ^
```

Comment (strega-nil) [reply ports/bond/CONTROL:6]: ah, this needs to add `-Wno-error=deprecated-declarations` to CXXFLAGS instead.

**Comment by JackBoosY** [comment]
Temporary close this PR. Will reopen this when I have enough time.

**Pull request closed without merging**